ci/docker.sh

```diff
@@ -29,46 +32,95 @@ fi
-    log_info "Publishing docker image: ${docker_image_build_name}:${publish_tag}"
-    docker tag "${docker_hub_org}/${docker_image_build_name}:${docker_tag_full}" "index.docker.io/${docker_hub_org}/${docker_image_build_name}:${publish_tag}"
-    docker push "index.docker.io/${docker_hub_org}/${docker_image_build_name}:${publish_tag}"
+    log_info "Publishing docker image to Docker Hub: ${docker_hub_org}/${docker_image_build_name}:${publish_tag}"
+    docker tag "${docker_image_build_name}:${docker_tag_full}" "${docker_hub_org}/${docker_image_build_name}:${publish_tag}"
+    docker push "${docker_hub_org}/${docker_image_build_name}:${publish_tag}"
   done
+
+  # Publish to Private Repository if specified
+  if [[ -n "${private_docker_repo:-}" ]]; then
+    log_info "=== Publishing Docker image(s) to Private Repository: ${private_docker_repo} ==="
+    if [[ -z "${DOCKER_HUB_TOKEN:-}" || -z "${DOCKER_HUB_USERNAME:-}" ]]; then
+      fn_die "ERROR: DOCKER_HUB_USERNAME and DOCKER_HUB_TOKEN must be set for private repository operations."
+    fi
+    echo "${DOCKER_HUB_TOKEN}" | docker login -u "${DOCKER_HUB_USERNAME}" --password-stdin "${private_docker_repo}"
+
+    for publish_tag in "${publish_tags[@]}"; do
+      log_info "Publishing docker image to Private Repository: ${private_docker_repo}:${publish_tag}"
+      docker tag "${docker_image_build_name}:${docker_tag_full}" "${private_docker_repo}:${publish_tag}"
+      docker push "${private_docker_repo}:${publish_tag}"
+    done
+  fi
+
+elif [[ "${is_a_release}" == "false" && "${dev_release}" == "true" && -n "${private_docker_repo:-}" ]]; then
+  # IS_A_RELEASE is false, but PRIVATE_DOCKER_REPO is set and DEV_RELEASE is true
+  if [[ -n "${commit_hash:-}" ]]; then
+    docker_tag_full="${commit_hash}"
+  else
+    docker_tag_full="dev"
+  fi
+  publish_tags=("${docker_tag_full}")
+
+  # Build and publish images to PRIVATE_DOCKER_REPO only
+  # Build the Docker image
+  log_info "=== Building Docker image: ${docker_image_build_name}:${docker_tag_full} ==="
+  docker build --build-arg PROFILE=production -f "${docker_file_path}" -t "${docker_image_build_name}:${docker_tag_full}" .
+
+  # Login to Private Docker Repository
+  log_info "=== Publishing Docker image(s) to Private Docker Repository: ${private_docker_repo} ==="
+  if [[ -z "${DOCKER_HUB_TOKEN:-}" || -z "${DOCKER_HUB_USERNAME:-}" ]]; then
+    fn_die "ERROR: DOCKER_HUB_USERNAME and DOCKER_HUB_TOKEN must be set for private repository operations."
+  fi
+  echo "${DOCKER_HUB_TOKEN}" | docker login -u "${DOCKER_HUB_USERNAME}" --password-stdin
+
+  # Tag and push to Private Repository
+  for publish_tag in "${publish_tags[@]}"; do
+    log_info "Publishing docker image to Private Repository: ${private_docker_repo}:${publish_tag}"
+    docker tag "${docker_image_build_name}:${docker_tag_full}" "${private_docker_repo}:${publish_tag}"
+    docker push "${private_docker_repo}:${publish_tag}"
+  done
```

Comment:
Again repetition of the previous blocks

Reply:
In this case this is necessary to be able to build and publish if the script is trigger by main merges

Reply:
I see, I think we can introduce new logic under setup_env.sh where we will be checking if BRANCH = main and checking other requirements setting for example IS_A_RELEASE=true and RELEASE_FROM_MAIN=true, then here we will be adding and extra check in the previous block along with DEV,TEST,PROD release and etc. Let me know if it makes sense